lib/schema_tools/modules/attributes.rb

```diff
@@ -84,25 +85,26 @@ def from_hash(hash, obj=nil)
           obj ||= new
           hash.each do |key, val|
             next unless obj.respond_to?(key)
+            conv_val = nil
             # set values to raw schema attributes, even if there are no setters
             # assuming this objects comes from a remote site
             field_props = self.schema['properties']["#{key}"]
-            conv_val = if val.nil?
-                         val
-                       elsif field_props['type'] == 'string'
-                         if field_props['format'] == 'date'
-                          Date.parse(val) # or be explicit? Date.strptime('2001-02-03', '%Y-%m-%d')
-                         elsif field_props['format'] == 'date-time'
-                           Time.parse(val) # vs Time.strptime
-                         else
-                          "#{val}"
-                         end
-                       elsif field_props['type'] == 'integer'
-                         val.to_i
-                       else # rely on preceding call e.g from_json for boolean, number
-                         val
-                       end
-                      # TODO if val is a hash / array => look for nested class & cast
+            field_type = field_props['type']
+            unless val.nil?
+              case field_type
```

Comment:
This is the actual factory. I wonder if it's worth breaking it out into it's own object.